src/crc32c/__init__.py

```diff
@@ -1 +1,32 @@
+import copy
+import base64
+
 from ._crc32c import big_endian, crc32, crc32c, hardware_based
+
+
+class Crc32cHash:
+    @property
+    def digest_size(self):
+        return 4
+
+    @property
+    def name(self):
+        return "crc32c"
+
+    def __init__(self, data):
+        self._checksum = crc32c(data)
+
+    def update(self, __data):
+        self._checksum = crc32c(__data, self._checksum)
+
+    def base64(self):
+        return base64.b64encode(self.digest()).decode(encoding="ascii")
```

Comment:
I went for a bit of a gamble here, I have to say.

I'm aware that the `hashlib` interfaces are *not* exposing this. But, in the actual, "wild" world, I feel like the `base64`-encoded value of a hash is pretty much one of the standard ways of exchanging it, at least when you're speaking to some kind of modern web API interfaces (AWS S3 as one, very big example). Also, many online CRC32(C) calculators offer to give the hash value in either hex- or base64-encoding. But I'm biased here, since I, in a professional context, don't have a need for the `int`, `digest`, or `hexdigest` representation but purely the `base64` one.

We're not *strictly* adhearing to the `hashlib` interface anyways, as can be seen for example in all of these methods I didn't implement: https://github.com/python/typeshed/blob/main/stdlib/hashlib.pyi#L74-L80. Also, I did not implement `block_size` because I'm unsure what value this has: https://github.com/python/typeshed/blob/main/stdlib/hashlib.pyi#L54-L55 I'm not even sure that the block size is constant across all different implementations of CRC32C in this library? If we wanted to stay extremely consitent with the `hashlib` interface, the class `Crc32cHash` would just inherit from `hashlib._hash` [or `_hashlib.HASH`] in the stub file, but I decided against it as I saw little value in it. This is also because we'd then would have to implement means to convert to MD5, SHA[X], ...

Let me know what you think, it's your decision :)

Reply:
Yes, I'm aware this interface is haslib-esque, so I wouldn't aim to implement it like-for-like. I also did have a brief look before my previous round of comments to see if there was a common base class, which would have helped clarified users what the class is meant to do, but didn't see one publicly documented. So no dramas there.

I hadn't even realised about the missing `block_size`. Yes, different underlying implementations of the algorithm might be implemented by processing more than one byte at a time, but those are optimisations -- the algorithm itself never adds padding. Fundamentally the algorithm's block size is `1`. Feel free to add this (or not :P)

I guess I don't have the same experience about sharing checksums as you do. I usually see them in command line tools or databases, in which cases I see hexadecimals, or even integers, this is the first time I hear about base64 representation being use wildly. I'm not saying I have the better perspective of things, just sharing my experience. Anyway, I'll stick to NOT having the base64 encoding here. It's a very simple operation on the user side, and OTOH it might become a source of complexity for us if we add it because of the different dialects/altchars/sizes. Let's see how popular this new hash class becomes, maybe more people mention they'd like to see it added in the same fashion.

Reply:
Added `block_size`, set it to 1 :shrug: 

Removed everything around `base64`.